code/datums/components/sticky.dm

```diff
@@ -0,0 +1,102 @@
+// kinda like duct tape but not shit (just kidding, its shit but in a different way)
+/datum/component/sticky
+	var/atom/attached_to
+	var/icon/overlay
+
+/datum/component/sticky/Initialize(obj/item/I)
+	if(!isitem(parent))
+		stack_trace("/datum/component/sticky's parent is not an item, its [parent.type]")
```

Comment:
`return COMPONENT_INCOMPATIBLE` after this.

Reply:
stack_trace is the non-crashing version of CRASH() iirc, it shouldn't stop it